keycloak_openid_client_permissions and keycloak_users_permissions require to explicitely define all scopes

# Issue

A `keycloak_openid_client_permissions` without all the scopes defined leads to a crash at refresh

```
resource "keycloak_openid_client_permissions" "example" {
  realm_id  = "MyRealm"
  client_id = "MyClientId"

  token_exchange_scope {
    policies = [
      "MyPolicyId,
    ]
    description       = "A description"
    decision_strategy = "UNANIMOUS"
  }
}
```

Similarly, a  `keycloak_users_permissions` without all the scopes defined leads to a crash at refresh

```
resource "keycloak_users_permissions" "example" {
  realm_id  = "MyRealm"

  manage_scope {
    policies = [
      "MyPolicyId,
    ]
    description       = "A description"
    decision_strategy = "UNANIMOUS"
  }
}
```


**Traceback**

(keycloak_openid_client_permissions)
```
Stack trace from the terraform-provider-keycloak_v3.3.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 232 [running]:
github.com/mrparkers/terraform-provider-keycloak/provider.resourceKeycloakUsersPermissionsRead(0xc00081cf80, 0x102e2e0, 0xc0007b8ae0, 0x178fa10, 0xc000808400)
        github.com/mrparkers/terraform-provider-keycloak/provider/resource_keycloak_users_permissions.go:127 +0xcff
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00065b5e0, 0x11d42e8, 0xc0006a7440, 0xc00081cf80, 0x102e2e0, 0xc0007b8ae0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00065b5e0, 0x11d42e8, 0xc0006a7440, 0xc00072d420, 0x102e2e0, 0xc0007b8ae0, 0xc00037c710, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000cde0, 0x11d42e8, 0xc0006a7440, 0xc0006a7480, 0xc0006a7440, 0x40b965, 0xfa9040)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0005e8860, 0x11d4390, 0xc0006a7440, 0xc000677f20, 0xc0005e8860, 0xc000797620, 0xc00016dba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0xfe6d40, 0xc0005e8860, 0x11d4390, 0xc000797620, 0xc000677ec0, 0x0, 0x11d4390, 0xc000797620, 0xc00071a180, 0x171)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004e41c0, 0x11dcb58, 0xc000102900, 0xc000394c00, 0xc000796930, 0x174fed0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0004e41c0, 0x11dcb58, 0xc000102900, 0xc000394c00, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0004c0190, 0xc0004e41c0, 0x11dcb58, 0xc000102900, 0xc000394c00)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-keycloak_v3.3.0 plugin crashed!
```

(keycloak_users_permissions)
```
Stack trace from the terraform-provider-keycloak_v3.3.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 157 [running]:
github.com/mrparkers/terraform-provider-keycloak/provider.resourceKeycloakUsersPermissionsRead(0xc0000f6b80, 0x102e2e0, 0xc0003864e0, 0x178fa10, 0xc000680000)
        github.com/mrparkers/terraform-provider-keycloak/provider/resource_keycloak_users_permissions.go:127 +0xcff
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0002175e0, 0x11d42e8, 0xc000317680, 0xc0000f6b80, 0x102e2e0, 0xc0003864e0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0002175e0, 0x11d42e8, 0xc000317680, 0xc0001481c0, 0x102e2e0, 0xc0003864e0, 0xc00011c3c0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000304618, 0x11d42e8, 0xc000317680, 0xc0003176c0, 0xc000317680, 0x40b965, 0xfa9040)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc00038a420, 0x11d4390, 0xc000317680, 0xc00068ca80, 0xc00038a420, 0xc0004fb290, 0xc00047bba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0xfe6d40, 0xc00038a420, 0x11d4390, 0xc0004fb290, 0xc00068ca20, 0x0, 0x11d4390, 0xc0004fb290, 0xc00058c780, 0x171)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003356c0, 0x11dcb58, 0xc000643e00, 0xc00037e600, 0xc0001d1530, 0x174fed0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0003356c0, 0x11dcb58, 0xc000643e00, 0xc00037e600, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0001221b0, 0xc0003356c0, 0x11dcb58, 0xc000643e00, 0xc00037e600)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-keycloak_v3.3.0 plugin crashed!
```

# How to reproduce

- Paste the example provided above, and replace it with correct values
- Apply the change
- Plan

# Expected

A plan should be provided with no detected changes to `keycloak_openid_client_permissions.example` or `keycloak_users_permissions`

# Context

Terraform version: 1.0.4
Provider version: 3.3.0

# Actual

The provider crashes with the traceback provided above

# Workaround

Defining all the scopes (before **any** apply, state corrupted otherwise) solves the issue

```
resource "keycloak_openid_client_permissions" "example" {
  realm_id  = "MyRealm"
  client_id = "MyClientId"

  view_scope {
    policies          = []
    description       = "Unused"
    decision_strategy = "UNANIMOUS"
  }

  manage_scope {
    policies          = []
    description       = "Unused"
    decision_strategy = "UNANIMOUS"
  }

  map_roles_scope {
    policies          = []
    description       = "Unused"
    decision_strategy = "UNANIMOUS"
  }

  token_exchange_scope {
    policies = [
      "MyPolicyId,
    ]
    description       = "A description"
    decision_strategy = "UNANIMOUS"
  }
}
```

```
resource "keycloak_users_permissions" "example" {
  realm_id  = "MyRealm"

  view_scope {
    policies          = []
    description       = ""
    decision_strategy = "UNANIMOUS"
  }

  manage_scope {
    policies          = []
    description       = ""
    decision_strategy = "UNANIMOUS"
  }

  map_roles_scope {
    policies          = []
    description       = ""
    decision_strategy = "UNANIMOUS"
  }

  manage_group_membership_scope {
    policies          = []
    description       = ""
    decision_strategy = "UNANIMOUS"
  }

  user_impersonated_scope {
    policies          = []
    description       = ""
    decision_strategy = "UNANIMOUS"
  }

  impersonate_scope {
    policies          = ["MyPolicy"]
    description       = ""
    decision_strategy = "UNANIMOUS"
  }
}
```



(Hope my issue template is complete enough!)

--- a/keycloak/openid_client_permissions.go
+++ b/keycloak/openid_client_permissions.go
@@ -9,11 +9,11 @@ type OpenidClientPermissionsInput struct {
 }
 
 type OpenidClientPermissions struct {
-	RealmId          string                 `json:"-"`
-	ClientId         string                 `json:"-"`
-	Enabled          bool                   `json:"enabled"`
-	Resource         string                 `json:"resource"`
-	ScopePermissions map[string]interface{} `json:"scopePermissions"`
+	RealmId          string            `json:"-"`
+	ClientId         string            `json:"-"`
+	Enabled          bool              `json:"enabled"`
+	Resource         string            `json:"resource"`
+	ScopePermissions map[string]string `json:"scopePermissions"`
 }
 
 func (keycloakClient *KeycloakClient) EnableOpenidClientPermissions(realmId, clientId string) error {

--- a/keycloak/users_permissions.go
+++ b/keycloak/users_permissions.go
@@ -9,10 +9,10 @@ type UsersPermissionsInput struct {
 }
 
 type UsersPermissions struct {
-	RealmId          string                 `json:"-"`
-	Enabled          bool                   `json:"enabled"`
-	Resource         string                 `json:"resource"`
-	ScopePermissions map[string]interface{} `json:"scopePermissions"`
+	RealmId          string            `json:"-"`
+	Enabled          bool              `json:"enabled"`
+	Resource         string            `json:"resource"`
+	ScopePermissions map[string]string `json:"scopePermissions"`
 }
 
 func (keycloakClient *KeycloakClient) EnableUsersPermissions(realmId string) error {

--- a/provider/resource_keycloak_openid_client_permissions.go
+++ b/provider/resource_keycloak_openid_client_permissions.go
@@ -107,43 +107,43 @@ func resourceKeycloakOpenidClientPermissionsUpdate(data *schema.ResourceData, me
 	}
 
 	if viewScope, ok := data.GetOk("view_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["view"].(string), viewScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["view"], viewScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if manageScope, ok := data.GetOk("manage_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["manage"].(string), manageScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["manage"], manageScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if configureScope, ok := data.GetOk("configure_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["configure"].(string), configureScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["configure"], configureScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if mapRolesScope, ok := data.GetOk("map_roles_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles"].(string), mapRolesScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles"], mapRolesScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if mapRolesClientsScope, ok := data.GetOk("map_roles_client_scope_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-client-scope"].(string), mapRolesClientsScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-client-scope"], mapRolesClientsScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if mapRolesCompositeScope, ok := data.GetOk("map_roles_composite_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-composite"].(string), mapRolesCompositeScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-composite"], mapRolesCompositeScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if tokenExchangeScope, ok := data.GetOk("token_exchange_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["token-exchange"].(string), tokenExchangeScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["token-exchange"], tokenExchangeScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
@@ -179,43 +179,43 @@ func resourceKeycloakOpenidClientPermissionsRead(data *schema.ResourceData, meta
 	data.Set("enabled", openidClientPermissions.Enabled)
 	data.Set("authorization_resource_server_id", realmManagementClient.Id)
 
-	if viewScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["view"].(string)); err == nil && viewScope != nil {
+	if viewScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["view"]); err == nil && viewScope != nil {
 		data.Set("view_scope", []interface{}{viewScope})
 	} else if err != nil {
 		return err
 	}
 
-	if manageScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["manage"].(string)); err == nil && manageScope != nil {
+	if manageScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["manage"]); err == nil && manageScope != nil {
 		data.Set("manage_scope", []interface{}{manageScope})
 	} else if err != nil {
 		return err
 	}
 
-	if mapRolesScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["configure"].(string)); err == nil && mapRolesScope != nil {
+	if mapRolesScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["configure"]); err == nil && mapRolesScope != nil {
 		data.Set("configure_scope", []interface{}{mapRolesScope})
 	} else if err != nil {
 		return err
 	}
 
-	if manageGroupMembershipScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles"].(string)); err == nil && manageGroupMembershipScope != nil {
+	if manageGroupMembershipScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles"]); err == nil && manageGroupMembershipScope != nil {
 		data.Set("map_roles_scope", []interface{}{manageGroupMembershipScope})
 	} else if err != nil {
 		return err
 	}
 
-	if impersonateScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-client-scope"].(string)); err == nil && impersonateScope != nil {
+	if impersonateScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-client-scope"]); err == nil && impersonateScope != nil {
 		data.Set("map_roles_client_scope_scope", []interface{}{impersonateScope})
 	} else if err != nil {
 		return err
 	}
 
-	if userImpersonatedScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-composite"].(string)); err == nil && userImpersonatedScope != nil {
+	if userImpersonatedScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["map-roles-composite"]); err == nil && userImpersonatedScope != nil {
 		data.Set("map_roles_composite_scope", []interface{}{userImpersonatedScope})
 	} else if err != nil {
 		return err
 	}
 
-	if tokenExchangeScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["token-exchange"].(string)); err == nil && tokenExchangeScope != nil {
+	if tokenExchangeScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, openidClientPermissions.ScopePermissions["token-exchange"]); err == nil && tokenExchangeScope != nil {
 		data.Set("token_exchange_scope", []interface{}{tokenExchangeScope})
 	} else if err != nil {
 		return err

--- a/provider/resource_keycloak_openid_client_permissions_test.go
+++ b/provider/resource_keycloak_openid_client_permissions_test.go
@@ -61,7 +61,7 @@ func testAccCheckKeycloakOpenidClientPermissionExists(resourceName string) resou
 			return fmt.Errorf("computed authorizationResourceServerId %s was not equal to %s (the id of the realm-management client)", authorizationResourceServerId, realmManagementId)
 		}
 
-		authzClientView, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["view"].(string))
+		authzClientView, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["view"])
 		if err != nil {
 			return err
 		}

--- a/provider/resource_keycloak_users_permissions.go
+++ b/provider/resource_keycloak_users_permissions.go
@@ -3,6 +3,7 @@ package provider
 import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"log"
 )
 
 func resourceKeycloakUsersPermissions() *schema.Resource {
@@ -66,37 +67,37 @@ func resourceKeycloakUsersPermissionsUpdate(data *schema.ResourceData, meta inte
 	}
 
 	if viewScope, ok := data.GetOk("view_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["view"].(string), viewScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["view"], viewScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if manageScope, ok := data.GetOk("manage_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage"].(string), manageScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage"], manageScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if mapRolesScope, ok := data.GetOk("map_roles_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["map-roles"].(string), mapRolesScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["map-roles"], mapRolesScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if manageGroupMembershipScope, ok := data.GetOk("manage_group_membership_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage-group-membership"].(string), manageGroupMembershipScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage-group-membership"], manageGroupMembershipScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if impersonateScope, ok := data.GetOk("impersonate_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["impersonate"].(string), impersonateScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["impersonate"], impersonateScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
 	}
 	if userImpersonatedScope, ok := data.GetOk("user_impersonated_scope"); ok {
-		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["user-impersonated"].(string), userImpersonatedScope.(*schema.Set))
+		err := setOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["user-impersonated"], userImpersonatedScope.(*schema.Set))
 		if err != nil {
 			return err
 		}
@@ -119,42 +120,48 @@ func resourceKeycloakUsersPermissionsRead(data *schema.ResourceData, meta interf
 		return handleNotFoundError(err, data)
 	}
 
+	if !usersPermissions.Enabled {
+		log.Printf("[WARN] Removing resource with id %s from state as it no longer enabled", data.Id())
+		data.SetId("")
+		return nil
+	}
+
 	data.SetId(usersPermissions.RealmId)
 	data.Set("realm_id", usersPermissions.RealmId)
 	data.Set("enabled", usersPermissions.Enabled)
 	data.Set("authorization_resource_server_id", realmManagementClient.Id)
 
-	if viewScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["view"].(string)); err == nil && viewScope != nil {
+	if viewScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["view"]); err == nil && viewScope != nil {
 		data.Set("view_scope", []interface{}{viewScope})
 	} else if err != nil {
 		return err
 	}
 
-	if manageScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage"].(string)); err == nil && manageScope != nil {
+	if manageScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage"]); err == nil && manageScope != nil {
 		data.Set("manage_scope", []interface{}{manageScope})
 	} else if err != nil {
 		return err
 	}
 
-	if mapRolesScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["map-roles"].(string)); err == nil && mapRolesScope != nil {
+	if mapRolesScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["map-roles"]); err == nil && mapRolesScope != nil {
 		data.Set("map_roles_scope", []interface{}{mapRolesScope})
 	} else if err != nil {
 		return err
 	}
 
-	if manageGroupMembershipScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage-group-membership"].(string)); err == nil && manageGroupMembershipScope != nil {
+	if manageGroupMembershipScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage-group-membership"]); err == nil && manageGroupMembershipScope != nil {
 		data.Set("manage_group_membership_scope", []interface{}{manageGroupMembershipScope})
 	} else if err != nil {
 		return err
 	}
 
-	if impersonateScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["impersonate"].(string)); err == nil && impersonateScope != nil {
+	if impersonateScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["impersonate"]); err == nil && impersonateScope != nil {
 		data.Set("impersonate_scope", []interface{}{impersonateScope})
 	} else if err != nil {
 		return err
 	}
 
-	if userImpersonatedScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["user-impersonated"].(string)); err == nil && userImpersonatedScope != nil {
+	if userImpersonatedScope, err := getOpenidClientScopePermissionPolicy(keycloakClient, realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["user-impersonated"]); err == nil && userImpersonatedScope != nil {
 		data.Set("user_impersonated_scope", []interface{}{userImpersonatedScope})
 	} else if err != nil {
 		return err

--- a/provider/resource_keycloak_users_permissions_test.go
+++ b/provider/resource_keycloak_users_permissions_test.go
@@ -68,7 +68,7 @@ func testAccCheckKeycloakUsersPermissionExists(resourceName string) resource.Tes
 		viewScopeDescription := rs.Primary.Attributes["view_scope.0.description"]
 		viewScopeDecisionStrategy := rs.Primary.Attributes["view_scope.0.decision_strategy"]
 
-		authzClientView, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["view"].(string))
+		authzClientView, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["view"])
 		if err != nil {
 			return err
 		}
@@ -86,7 +86,7 @@ func testAccCheckKeycloakUsersPermissionExists(resourceName string) resource.Tes
 			return fmt.Errorf("decision strategy %s was not equal to %s", authzClientView.DecisionStrategy, viewScopeDecisionStrategy)
 		}
 
-		authzClientManage, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["manage"].(string))
+		authzClientManage, err := keycloakClient.GetOpenidClientAuthorizationPermission(permissions.RealmId, realmManagementId, permissions.ScopePermissions["manage"])
 		if err != nil {
 			return err
 		}

